feat: add templates showcase pages

GuiBibeau opened the pull request:
## Summary

Restructured templates feature to align with the monorepo architecture by moving it to a standalone app in /apps/templates.

## What Changed

- Moved templates from apps/web/src/app/[locale]/templates to apps/templates
- Removed packages/templates workspace package that was causing architectural confusion
- Created self-contained Next.js app with own configuration, routing, and assets
- Fixed TypeScript errors in packages/ui-chrome/src/link.tsx
- Added templates routes to main site sitemap for SEO
- Removed templates-specific translations and assets from apps/web
- Added rewrite configuration placeholder for production deployment

## Architecture

The templates app follows the new Turbo monorepo structure where standalone features live in /apps/* rather than being embedded in apps/web. The app:

- Uses shared packages from /packages/* (ui-chrome, ui, i18n)
- Can be deployed separately from the main site
- Accessed via Next.js rewrites from /templates on solana.com
- Includes own build, dev server (port 3001), and deployment configuration

## Deployment Strategy

The templates app is designed to be deployed as a separate application. The main web app will rewrite /templates routes to the templates app deployment URL via Next.js rewrites configuration in production.

<img width="1448" height="907" alt="Screenshot 2025-10-23 at 5 04 50 PM" src="https://github.com/user-attachments/assets/7197a990-2098-4ee5-a200-62ee854b8816" />

h4rkl reviewed (commented):
Just noting this here since we’ve already discussed it — posting for visibility so we can resolve and avoid it getting merged accidentally.

Templates should live in their own folder under /apps, since they’re essentially distinct apps. The new Turbo setup is structured so we don’t add anything new inside apps/web, and can gradually move standalone sections into /apps/<app_name> to reduce apps/web file count, technical debt, and complexity.

That’s also why shared header/footer packages used to be a blocker — they couldn’t be shared easily between apps but now live in /packages/ui-chrome. All other shared packages (like i18n, linting, etc.) live under /packages 🎉

GuiBibeau commented:
working now on simplifying the stuff duplicated

GuiBibeau commented:
@h4rkl updates:
- fully changed to be a standalone app
- tapped into the shared i18n stuff
- added to the sitemap 
- moved over our components into the new next.js app. 

One i was not able to get done within reasonable time is a cleaner font shred package. I created an issue to follow up
